src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.Tests/CallSiteTests.cs

```diff
@@ -264,6 +264,32 @@ public void DoesNotThrowWhenServiceIsUsedAsEnumerableAndNotInOneCallSite()
             Assert.NotNull(compileCallSite);
         }
 
+        [Fact]
+        public void NoServiceCallsite_DefaultValueNull_DoesNotThrow()
+        {
+            var descriptors = new ServiceCollection();
+            descriptors.AddTransient<ServiceG>();
+
+            var provider = descriptors.BuildServiceProvider(ServiceProviderMode.ILEmit);
```

Comment:
Should we test all the service provider modes? i.e. Use a Theory and pass in `ServiceProviderMode` to the test.